### PR TITLE
Clarify label_replace()

### PR DESCRIPTION
**Pull request by nikopen**
It's a really useful function but I find the current docs too high level and "scientific" for people not so averse with monitoring to understand. The two sentences added might be implied from the already existing doc, but the conclusion is not straightforward.

It's best to simplify docs and terms in order for adoption to grow.

### CODE REVIEW

**Comment by nikopen** [comment]
I'm not sure what sign-off is and why it's required. 

I used the github web edit to make my change.

**Comment by SuperQ** [comment]
@nikopen Sign-off is required for legal project contribution reasons. See https://github.com/probot/dco#how-it-works and https://developercertificate.org/

File: docs/querying/functions.md [hunk 1]
```diff
@@ -258,6 +258,15 @@ timeseries is returned with the label `dst_label` replaced by the expansion of
 second etc. If the regular expression doesn't match then the timeseries is
 returned unchanged.
 
+If `src_label` and `dst_label` are the same, the label's value will be altered based on the regex.
```
[the remark this answers is not in the document]
Comment (nikopen) [reply docs/querying/functions.md]: re:duplicating, the whole section of label_replace is really not easy to grasp (someone could argue the majority of the docs share the same concern, but it's a start). The existing paragraph might be compacted enough to contain all mechanical meaning, but it doesn't really translate easily on what a user understands this function is actually meant to do. 

Yes it's reference documentation, but there is no alternative. Users have to resort to googling to older blogposts and _maybe_ finding what they need.

Is there any harm to simplify docs wherever possible and give simpler examples?

File: docs/querying/functions.md [hunk 2]
```diff
@@ -258,6 +258,15 @@ timeseries is returned with the label `dst_label` replaced by the expansion of
 second etc. If the regular expression doesn't match then the timeseries is
 returned unchanged.
 
+If `src_label` and `dst_label` are the same, the label's value will be altered based on the regex. 
+If they are different, a new `dst_label` will be added.
+
+This example will add a new label "container_name" having the same values as "container" as (.+) selects everything:
```
[the remark this answers is not in the document]
Comment (nikopen) [reply docs/querying/functions.md]: replaced "as (.+) selects everything" with "(if "container" is not empty)".

I'd argue that the existing example is not clear enough.

[the remark this answers is not in the document]
Comment (brian-brazil) [reply docs/querying/functions.md]: This is about label_replace though, not how regexes work.

[the remark this answers is not in the document]
Comment (nikopen) [reply docs/querying/functions.md]: It's a very common use case which has great usability in the k8s world, as you need to link docker's "container_name" with k8s "container" or "pod_name". 

It gives you a straightforward common use case and I can't see any argument against it.

If Prometheus was targeted at mathematicians then yes, it's a perfect reference doc. But it's also aimed at many people who haven't built monitoring before, and making it easier to grasp can make monitoring great again.

**Comment by oseiberts11** [comment]
You can also use `label_replace` to delete labels which is also not documented. Strangely enough this is done by using "" as a new value. Annoyingly I can't find the place any more where I found that (some PR proposing some additions/changes).

**Comment by roidelapluie** [comment]
> You can also use `label_replace` to delete labels which is also not documented. Strangely enough this is done by using "" as a new value. Annoyingly I can't find the place any more where I found that (some PR proposing some additions/changes).

We should not document that. This is a side effect to the fact that empty labels are the same as no label.

The recommended way to remove labels is to tell prometheus what to do with duplicates. e.g.:

max without(removedlabel) my  metric

**Comment by oseiberts11** [comment]
Hm, that looks even weirder to me, to be honest. I have also seen the same with using `sum`. Why not an explicit `remove_label` function then? There should be a single canonical way to do it, that would be much better imo.

**Comment by roidelapluie** [comment]
removing labels with label_replace is a cosmetic change, it is better to use `on`, `ignoring`, `without` etc. The cosmetic part should be done by the graphing solution.

**Comment by codesome** [comment]
We discussed during our bug scrub and decided to close this as it's duplicating the information incorrectly. We hope to have a new section in the docs soon which will explain the functions better.

**Pull request closed without merging**